qiskit/quantum_info/states/quantum_state.py

```diff
@@ -55,6 +56,129 @@ def __repr__(self):
         return '{}({}, dims={})'.format(
             self.rep, self.data, self._dims)
 
+    def _repr_latex_(self):
+        """
+        Generate a LaTeX representation of the Quantum State.
+
+        This file is part of QuTiP: Quantum Toolbox in Python.
+
+        Copyright (c) 2011 and later, Paul D. Nation and Robert J. Johansson.
+        All rights reserved.
+        """
+        shape = self._data.shape
```

Comment:
This assumes that `state._data` is a numpy array, which is true for Statevector and DensityMatrix classes, but shouldn't be assumed in the base class I don't think. Can you move this to the specific classes?

Reply:
Is there some other inputs than numpy arrays planned here?  Sparse would also be easy to handle. I can move though. Will duplicate the code.